include/cvc5/cvc5_proof_rule.h

```diff
@@ -2702,6 +2702,21 @@ enum ENUM(ProofRewriteRule)
    * \endverbatim
    */
   EVALUE(QUANT_DT_SPLIT),
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Quantifiers -- Macro datatype variable expand **
+   *
+   * .. math::
+   *   (\forall Y x Z.\> F) = (\forall Y X_1 Z. F_1) \vee \cdots \vee (\forall Y X_n Z. F_n)
+   *
+   * where :math:`x` is of a datatype type with constructors
+   * :math:`C_1, \ldots, C_n`, where for each :math:`i = 1, \ldots, n`,
+   * :math:`F_i` is :math:`F \{ x \mapsto C_i(X_i) \}`, and
+   * :math:`F` entails :math:`\mathit{is}_c(x)` for some :math:`c`.
```

Comment:
> and :math:`F` entails :math:`\mathit{is}_c(x)` for some :math:`c`.

I don't understand this part. Doesn't the conclusion hold just from the type of `x`?

Reply:
Indeed, it does, although this indicates when the rewrite is performed.  In particular, we only apply this rewrite if the right hand side will simplify to a single disjunct, essentially.